### PR TITLE
fix for image component when try to request the remote image

### PR DESCRIPTION
**Pull request by Dileepnidi**
- Added check to set the headers
- verified available test app for  image components a.)ImageDataExpiryTime.js b.)getSize.js
- verified with XHR test apps.

### CODE REVIEW

**Review by ddesai96 — APPROVED** [review]

**Comment by chatrapathi06** [comment]
Change is looks okay to me.
2. Verify with all dependent modules/components.(Image/XHR) unit test APP.

**Review by chatrapathi06 — APPROVED** [review]
Verified for--> unit test app of image, XHR, getSize.

File: ReactSkia/sdk/CurlNetworking.cpp [hunk 1]
```diff
@@ -300,7 +300,8 @@ bool CurlNetworking::sendRequest(shared_ptr<CurlRequest> curlRequest, folly::dyn
     curl_easy_setopt(curl, CURLOPT_PROGRESSDATA, curlRequest.get() );
     curl_easy_setopt(curl, CURLOPT_PROGRESSFUNCTION, progressCallbackCurlWrapper);
   }
-  setHeaders(curlRequest, headers);
+  if(headers != nullptr)
```
Comment (pprabhu-ng) [inline ReactSkia/sdk/CurlNetworking.cpp]: please added braces for the if check

Comment (pprabhu-ng) [inline ReactSkia/sdk/CurlNetworking.cpp]: should we also add the check for "data" variable also , before calling preparePostRequest

File: ReactSkia/sdk/CurlNetworking.cpp [hunk 2]
```diff
@@ -300,7 +300,8 @@ bool CurlNetworking::sendRequest(shared_ptr<CurlRequest> curlRequest, folly::dyn
...
+    setHeaders(curlRequest, headers);
```
Comment (pprabhu-ng) [inline ReactSkia/sdk/CurlNetworking.cpp:304]: setHeaders is a public API of the sdk module, which means any client can call this API to set headers . so should the check be inside setHeaders function?

Comment (Dileepnidi) [reply ReactSkia/sdk/CurlNetworking.cpp:304]: setHeaders is private now

**Review by munezbn — APPROVED** [review]

**Review by pprabhu-ng — APPROVED** [review]

**Pull request merged**